# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1311636451 -7200
#      Tue Jul 26 01:27:31 2011 +0200
# Node ID edaf8e72070e4529208d02e264226d9a61f663a2
# Parent  fd87d6f7e18542a9d90b1af3cae8b1344760bdbe
Added urls for Agora and Octave Forge in Community Window.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -136,16 +136,30 @@ MainWindow::openEditor ()
 void
 MainWindow::openBugTrackerPage ()
 {
   QDesktopServices::
     openUrl (QUrl ("http://savannah.gnu.org/bugs/?group=octave"));
 }
 
 void
+MainWindow::openAgoraPage ()
+{
+  QDesktopServices::
+    openUrl (QUrl ("http://agora.panocha.org.mx/"));
+}
+
+void
+MainWindow::openOctaveForgePage ()
+{
+  QDesktopServices::
+    openUrl (QUrl ("http://octave.sourceforge.net/"));
+}
+
+void
 MainWindow::processSettingsDialogRequest ()
 {
   SettingsDialog settingsDialog (this, m_settingsFile);
   settingsDialog.exec ();
 }
 
 void
 MainWindow::closeEvent (QCloseEvent * closeEvent)
@@ -253,22 +267,27 @@ MainWindow::construct ()
   QMenu *workspaceMenu = menuBar ()->addMenu (tr ("Workspace"));
   QAction *loadWorkspaceAction = workspaceMenu->addAction (tr ("Load"));
   QAction *saveWorkspaceAction = workspaceMenu->addAction (tr ("Save"));
   workspaceMenu->addSeparator ();
   QAction *clearWorkspaceAction = workspaceMenu->addAction (tr ("Clear"));
 
   QMenu *communityMenu = menuBar ()->addMenu (tr ("Community"));
   QAction *reportBugAction = communityMenu->addAction (tr ("Report Bug"));
+  QAction *agoraAction = communityMenu->addAction (tr ("Agora"));
+  QAction *octaveForgeAction = communityMenu->addAction (tr ("Octave Forge"));
 
   connect (settingsAction, SIGNAL (triggered ()), this, SLOT (processSettingsDialogRequest ()));
   connect (exitAction, SIGNAL (triggered ()), this, SLOT (close ()));
   connect (alignWindowsAction, SIGNAL (triggered ()), this, SLOT (alignMdiWindows ()));
   connect (openEditorAction, SIGNAL (triggered ()), this, SLOT (openEditor ()));
   connect (reportBugAction, SIGNAL (triggered ()), this, SLOT (openBugTrackerPage ()));
+  connect (agoraAction, SIGNAL (triggered ()), this, SLOT (openAgoraPage ()));
+  connect (octaveForgeAction, SIGNAL (triggered ()), this, SLOT (openOctaveForgePage ()));
+
   connect (showWorkspaceAction, SIGNAL (toggled (bool)), m_variablesDockWidget, SLOT (setShown (bool)));
   connect (m_variablesDockWidget, SIGNAL (visibilityChanged (bool)), showWorkspaceAction, SLOT (setChecked (bool)));
   connect (showHistoryAction, SIGNAL (toggled (bool)), m_historyDockWidget, SLOT (setShown (bool)));
   connect (m_variablesDockWidget, SIGNAL (visibilityChanged (bool)), showHistoryAction, SLOT (setChecked (bool)));
   connect (showFileBrowserAction, SIGNAL (toggled (bool)), m_filesDockWidget, SLOT (setShown (bool)));
   connect (m_filesDockWidget, SIGNAL (visibilityChanged (bool)), showFileBrowserAction, SLOT (setChecked (bool)));
 
   setWindowTitle (QString (VERSION_STRING));
diff --git a/gui/src/MainWindow.h b/gui/src/MainWindow.h
--- a/gui/src/MainWindow.h
+++ b/gui/src/MainWindow.h
@@ -75,16 +75,18 @@ Q_OBJECT public:
   void openWebPage (QString url);
   void handleSaveWorkspaceRequest ();
   void handleLoadWorkspaceRequest ();
   void handleClearWorkspaceRequest ();
   void handleCommandDoubleClicked (QString command);
   void alignMdiWindows ();
   void openEditor ();
   void openBugTrackerPage ();
+  void openAgoraPage ();
+  void openOctaveForgePage ();
   void processSettingsDialogRequest ();
 
 protected:
   void closeEvent (QCloseEvent * closeEvent);
   void readSettings ();
   void writeSettings ();
 
 private:
